make.paws/tests/testthat/test_handlers_restxml.R

```diff
@@ -0,0 +1,692 @@
+    interface <- Structure(
+      BoolQuery = Scalar(type = "boolean", .attrs = list(location = "querystring", locationName = "bool-query"))
+    )
+    return(populate(args, interface))
+  }
+  input <- op_input7(
+    BoolQuery = TRUE
+  )
+  req <- new_request(svc, op7, input, NULL)
+  req <- build(req)
+  r <- req$http_request
+  expect_equal(build_url(r$url), "https://test/path?bool-query=true")
+})
+
+test_that("query string with bool (false)", {
+  op8 <- Operation(
+    name = "OperationName",
+    http_method = "GET",
+    http_path = "/path"
+  )
+  op_input8 <- function(BoolQuery) {
+    args <- list(BoolQuery = BoolQuery)
+    interface <- Structure(
+      BoolQuery = Scalar(type = "boolean", .attrs = list(location = "querystring", locationName = "bool-query"))
+    )
+    return(populate(args, interface))
+  }
+  input <- op_input8(
+    BoolQuery = FALSE
+  )
+  req <- new_request(svc, op8, input, NULL)
+  req <- build(req)
+  r <- req$http_request
+  expect_equal(build_url(r$url), "https://test/path?bool-query=false")
+})
+
+test_that("URI and query string parameters", {
+  op9 <- Operation(
+    name = "OperationName",
+    http_method = "GET",
+    http_path = "/2014-01-01/jobsByPipeline/{PipelineId}"
+  )
+  op_input9 <- function(Ascending, PageToken, PipelineId) {
+    args <- list(Ascending = Ascending, PageToken = PageToken, PipelineId = PipelineId)
+    interface <- Structure(
+      Ascending = Scalar(type = "string", .attrs = list(location = "querystring", locationName = "Ascending")),
+      PageToken = Scalar(type = "string", .attrs = list(location = "querystring", locationName = "PageToken")),
+      PipelineId = Scalar(type = "string", .attrs = list(location = "uri", locationName = "PipelineId"))
+    )
+    return(populate(args, interface))
+  }
+  input <- op_input9(
+    Ascending = "true",
+    PageToken = "bar",
+    PipelineId = "foo"
+  )
+  req <- new_request(svc, op9, input, NULL)
+  req <- build(req)
+  r <- req$http_request
+  expect_equal(build_url(r$url), "https://test/2014-01-01/jobsByPipeline/foo?Ascending=true&PageToken=bar")
+})
+
+
+#-------------------------------------------------------------------------------
+
+# XML tests
+
+test_that("Basic XML Case1", {
+  op_test <- Operation(name = "OperationName")
+  op_input_test <- function(OperationRequest) {
+    args <- list(OperationRequest = OperationRequest)
+    interface <- structure(list(
+      OperationRequest = structure(list(
+        Description = structure(logical(0), type = "string"),
+        Name = structure(logical(0), type = "string")
+      ), type = "structure", locationName = "OperationRequest")
+    ), type = "structure")
+    return(populate(args, interface))
+  }
+
+  input <- op_input_test(
+      OperationRequest = list(
+      Description = "bar",
+      Name = "foo")
+  )
+  req <- new_request(svc, op_test, input, NULL)
+  req <- build(req)
+  r <- req$body
+  expect_equal(r, xml_list_to_character(list(
```

Comment:
I think we should be copy and compare against the same output that the Go SDK test is comparing against, or at least a minimally-modified version of that. This compares against a tailored object put through a function that is also used by the function being tested.

Reply:
addressed in last commit. I also added in xml namespace support